src/convertToObject.js

```diff
@@ -6,7 +6,26 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  let strToObject = sourceString
+    .trim()
+    .split(';')
+    .filter((str) => str.trim());
+
+  strToObject = strToObject.map((str) => {
+    return str
+      .trim()
+      .split(':')
+      .map((s) => s.trim());
+  });
+
+  strToObject = Object.fromEntries(strToObject);
 
+  for (const key in strToObject) {
+    if (strToObject[key] === undefined) {
+      delete strToObject[key];
+    }
+  }
+
+  return strToObject;
```

Comment:
The variable name 'strToObject' is not descriptive of its contents or purpose. Consider using a more descriptive name that indicates the variable holds a style object, such as 'styleObject' or 'cssProperties'.